File: Data Structures & Algorithms/duplicate-integer/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums: number[]): boolean {
        const seen = new Set<number>();
        for (const num of nums) {
            if (seen.has(num)) return true; // Found a duplicate, exit early!
            seen.add(num);
        }
        return false;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-1.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        consecutiveOnes = [0]
        onesCounter = 0
        for num in nums:
            if num == 1: onesCounter += 1
            else:
                if onesCounter > 0: consecutiveOnes.append(onesCounter)
                onesCounter = 0
        if onesCounter > 0: consecutiveOnes.append(onesCounter)

        return max(consecutiveOnes)